src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectManagementService.cs

```diff
@@ -142,30 +145,43 @@ internal async Task HandleSearchRequest(SearchRequestParams requestParams, Reque
                     continue;
                 }
 
-                SearchableObjectTypeDescription desc = SearchableObjectTypeDescription.GetDescription(searchableObjectType);
+                // only schema-Scoped securableTypes support schema level search
+                if (!string.IsNullOrEmpty(requestParams.Schema) && !schemaTypes.Contains(searchableObjectType))
+                {
+                    continue;
+                }
 
-                if (requestParams.SearchText != null)
+                SearchableObjectTypeDescription desc = SearchableObjectTypeDescription.GetDescription(searchableObjectType);
+                
+                if (desc.IsDatabaseObject)
```

Comment:
there are so many branches here, I think this can be refactored to just one SearchableObject.Search() call.

Reply:
These branches are by nature, and we are still going to have similar branches in the refactored function (ultimately we need to construct different URN strings for each branch). If we want to keep code in objectmanagementservice clean I can probably move this chunk of code to utils class later.